module/VuFind/src/VuFind/Section/SectionService.php

```diff
@@ -0,0 +1,184 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Section
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Section;
+
+use VuFind\Config\Feature\ConfigSettingPropertiesInterface;
+use VuFind\Config\YamlReader;
+use VuFind\Exception\BadConfig;
+use VuFind\Exception\ConfigException;
+use VuFind\Navigation\PluginManager as NavigationPluginManager;
+use VuFind\Section\PluginManager as SectionPluginManager;
+
+use function is_string;
+
+/**
+ * Section service.
+ *
+ * @category VuFind
+ * @package  Section
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+class SectionService implements SectionServiceInterface
+{
+    /**
+     * Constructor.
+     *
+     * @param YamlReader              $yamlReader      YAML reader
+     * @param SectionPluginManager    $sectionPm       Section plugin manager
+     * @param NavigationPluginManager $navigationPm    Navigation plugin manager
+     * @param string                  $userLocale      User locale
+     * @param array                   $fallbackLocales Fallback locales
+     */
+    public function __construct(
+        protected YamlReader $yamlReader,
+        protected SectionPluginManager $sectionPm,
+        protected NavigationPluginManager $navigationPm,
+        protected string $userLocale,
+        protected array $fallbackLocales
+    ) {
+    }
+
+    /**
+     * Get section.
+     *
+     * @param string       $key    Section key in configuration
+     * @param array|string $config Configuration or configuration file name
+     *                             (optional)
+     *
+     * @return ?SectionInterface
+     * @throws ConfigException
+     * @throws BadConfig
+     */
+    public function getSection(
+        string $key,
+        array|string $config = self::DEFAULT_CONFIG_FILE
+    ): ?SectionInterface {
+        if (is_string($config)) {
+            $configFile = $this->getConfigFromFile($config);
+            $config = $this->getSectionConfig($key, $configFile);
+        }
+
+        if (!$type = $config['type'] ?? false) {
+            throw new BadConfig('Missing required setting: type');
+        }
+
+        $pluginManager = $this->sectionPm;
+
+        // Type specific configuration processing.
+        switch ($type) {
+            case 'navigation':
+                if (!$type = $config['plugin'] ?? false) {
```

Comment:
Maybe this is fine as-is, but it feels like more parentheses might be needed or at least helpful.